src/Symfony/Bundle/FrameworkBundle/HttpClient/TraceableHttpClient.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\HttpClient;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpCache\Esi;
+use Symfony\Component\HttpKernel\HttpCache\HttpCache as BaseHttpCache;
+use Symfony\Component\HttpKernel\HttpCache\Store;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+
+/**
+ * AAA
+ *
+ * @author Jérémy Romey <jeremy@free-agent.fr>
+ */
+class TraceableHttpClient implements HttpClientInterface
+{
+    protected $httpClient;
+    protected $traces = [];
+
+    /**
+     * @param HttpClientInterface $httpClient A HttpClientInterface instance
+     */
+    public function __construct(HttpClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        $response = $this->httpClient->request($method, $url, $options);
+
+        $this->traces[] = [
```

Comment:
i believe we cant simply decorate like this, due a lazy response stream. Which is now broken 100% of the calls, isnt it?

Reply:
Yes, only response objects should be kept until `HttpClientDataCollector::getTraces` is called, there should be the reading of the status codes, even in the templates so we're sure to read on next request if we don't show them in the toolbar. Only the request count should be shown there which is already the case for now.